### ISSUE TITLE
v1.7.3 error with proguard

### ISSUE THREAD

**Issue by tescaflown** [retrostreams/android-retrostreams#6]
Hello,

just update to 1.7.3, no problem in debug mode. But in release, i can't collect.

StreamSupport.stream(list).sorted(comparator).collect(Collectors.toCollection(supplier));

Problem arise when i collect (stream and sorted are ok). 

Collect give me this error:

java.lang.NoSuchMethodException: listFromTrustedArray [class [Ljava.lang.Object;]

I switched back to version 1.7.2, problem disappear. That's obviously some missing protection from proguard but what ?

**Comment by stefan-zobel** [comment]
In 1.7.3 there are two new package-private methods in **_java9.util.ImmutableCollections_**

```
 static <E> List<E> listFromTrustedArray(Object[])
 static <E> List<E> listFromTrustedArrayNullsAllowed(Object[])
```

which are called reflectively from **_java9.util.stream.Collectors_**.
The java.lang.method.Method instances for these two methods are initialized once in the Collectors static constructor,
that's why you get that Error when these methods are missing.

ProGuard must _not touch_ these two methods in java9.util.ImmutableCollections

**Comment by tescaflown** [comment]
thanks for your answer.  maybe should add a waning on the main page that's now we need to add a proguard rules to protect these 2 methods. Pretty sure, i won't be the only one to get an error :)

Have a good day

**Comment by Christophe31** [comment]
Not sure this issue should be closed, I didn't saw the fix in the project.

**Comment by cloewen8** [comment]
Had this occur today, not resolved in 1.7.4. Downgrading to 1.7.2 worked for me.
